converter.go

```diff
@@ -818,24 +959,137 @@ func mapStructuredArrayNativeArrowRows[T any](offsets []int32, rowIdx int, creat
+			return items.(*array.Boolean).Value(j), nil
+		})
+	case "fixed":
+		if higherPrecision && srcColumnMeta.Fields[1].Scale == 0 {
+			return mapStructuredMapNativeArrowRows(make(map[K]*big.Int), offsets, rowIdx, keyFunc, func(j int) (*big.Int, error) {
+				v, err := extractNumberFromArrow(&items, j, higherPrecision, srcColumnMeta.Fields[1].Scale)
+				if err != nil {
+					return nil, err
+				}
+				return v.(*big.Int), nil
+			})
+		} else if higherPrecision && srcColumnMeta.Fields[1].Scale != 0 {
+			return mapStructuredMapNativeArrowRows(make(map[K]*big.Float), offsets, rowIdx, keyFunc, func(j int) (*big.Float, error) {
+				v, err := extractNumberFromArrow(&items, j, higherPrecision, srcColumnMeta.Fields[1].Scale)
+				if err != nil {
+					return nil, err
+				}
+				return v.(*big.Float), nil
+			})
+		} else if !higherPrecision && srcColumnMeta.Fields[1].Scale == 0 {
+			return mapStructuredMapNativeArrowRows(make(map[K]int64), offsets, rowIdx, keyFunc, func(j int) (int64, error) {
+				v, err := extractNumberFromArrow(&items, int(j), higherPrecision, srcColumnMeta.Fields[1].Scale)
+				if err != nil {
+					return 0, err
+				}
+				return strconv.ParseInt(v.(string), 10, 64)
+			})
+		} else {
+			return mapStructuredMapNativeArrowRows(make(map[K]float64), offsets, rowIdx, keyFunc, func(j int) (float64, error) {
+				v, err := extractNumberFromArrow(&items, j, higherPrecision, srcColumnMeta.Fields[1].Scale)
+				if err != nil {
+					return 0, err
+				}
+				return strconv.ParseFloat(v.(string), 64)
+			})
+		}
+	case "real":
+		return mapStructuredMapNativeArrowRows(make(map[K]float64), offsets, rowIdx, keyFunc, func(j int) (float64, error) {
+			return items.(*array.Float64).Value(j), nil
+		})
+	case "binary":
+		return mapStructuredMapNativeArrowRows(make(map[K][]byte), offsets, rowIdx, keyFunc, func(j int) ([]byte, error) {
+			return items.(*array.Binary).Value(j), nil
+		})
+	case "date":
+		return mapStructuredMapNativeArrowRows(make(map[K]time.Time), offsets, rowIdx, keyFunc, func(j int) (time.Time, error) {
+			return arrowDateToValue(items.(*array.Date32), j).(time.Time), nil
+		})
+	case "time":
+		return mapStructuredMapNativeArrowRows(make(map[K]time.Time), offsets, rowIdx, keyFunc, func(j int) (time.Time, error) {
+			return arrowTimeToValue(items, j, srcColumnMeta.Fields[1].Scale).(time.Time), nil
+		})
+	case "timestamp_ltz", "timestamp_ntz", "timestamp_tz":
+		return mapStructuredMapNativeArrowRows(make(map[K]time.Time), offsets, rowIdx, keyFunc, func(j int) (time.Time, error) {
+			return *arrowSnowflakeTimestampToTime(items, getSnowflakeType(srcColumnMeta.Fields[1].Type), srcColumnMeta.Fields[1].Scale, j, loc), nil
+		})
+	}
+	return nil, errors.New("Unsupported map value: " + srcColumnMeta.Fields[1].Type)
+}
+
+func extractNumberFromArrow(values *arrow.Array, j int, higherPrecision bool, scale int) (snowflakeValue, error) {
```

Comment:
can we keep second arg as `rowIdx` instead of `j`? helps in understanding whats going on.

Reply:
Actually we can't because this is not `rowIdx`. This is offset indicator for specific row.

Reply:
Got it.